pkg/apis/eksctl.io/v1alpha4/types.go

```diff
@@ -378,15 +378,7 @@ type NodeGroup struct {
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
 
-	// TODO move to separate struct
-	// +optional
-	AllowSSH bool `json:"allowSSH"`
-	// +optional
-	SSHPublicKeyPath string `json:"sshPublicKeyPath,omitempty"`
-	// +optional
-	SSHPublicKey []byte `json:"SSHPublicKey,omitempty"` // TODO: right now it's kind of read-only, but one may wish to use key body in a config file so we will need recognise that
-	// +optional
-	SSHPublicKeyName string `json:"sshPublicKeyName,omitempty"`
+	SSH SSHConfig `json:"ssh"`
```

Comment:
Seems plausible to make this one a reference also, at least for consistency. Some of the fields above aren't references, but arguably that's just because we didn't get to refactor those yet.